instructor/common.py

```diff
@@ -68,19 +68,33 @@ def create_new_assignment(request):
 
 
 def assignment_setup(request):
+    error = ''
     if request.method == "POST":
-        # Have to give error for empty name
-        request.session['assignment_name'] = request.POST['name']
+        assignment_name = request.POST['name']
         if request.POST['based_on']:
             request.session['based_on'] = request.POST['based_on']
-
-        if 'continue' in request.POST:
-            return redirect('common_course_setup')
+        if assignment_name:
+            request.session['assignment_name'] = assignment_name
+            if 'continue' in request.POST:
+                return redirect('common_course_setup')
+        else:  # Have to give error for empty name
+            error = "*Empty assignment name"
+    else:
+        index = 1
+        found = False
+        while not found:
+            assignment_name = "Assignment" + str(index)
+            if models.Assignment.objects.filter(name=assignment_name).count() > 0:
+                index += 1
+            else:
+                found = True
```

Comment:
this loop might be correct but is not written in a pythonic way and has a serious problem:

1-(minor) in python you do not need a guardian variable to break a loop: you can use the `break` statement: this means you can get rid of the `found` variable
2-(serious) this while is not safe: in theory it can go on forever because there is no check based on the maximum amount of cycles to run: this means you have to check the maximum amount of loops it can run. Reading the code it seems that this number is equal to `models.Assignment.objects.count()`
3-(very minor) string concatenation should not be done with the `+` operator

Now looking at the code, I think you can use a different approach: if the maximum amount of loops you can run is `models.Assignment.objects.count()`, then you can write something like:
```
for index in xrange(models.Assignment.objects.count()):  # use range if this python should run in python 3
    if models.Assignment.objects.filter(name="Assignment{0}".format(index)).count():
        break;
```
This will stop at the first `index` that gives a count > 0 and you can reuse the variable later.

This is the first step, but I'm pretty sure you might even reduce the number of queries using the right filtering, maybe filtering by the `name` starting with `"Assignment"`, sorting in ascending mode and taking the first, but I'm not sure because I do not know the data in the database.